src/transformers/models/jamba/modeling_jamba.py

```diff
@@ -1383,6 +1410,17 @@ def _update_causal_mask(self, attention_mask, input_tensor, cache_position):
 
         return causal_mask
 
+    def _update_mamba_mask(self, attention_mask, cache_position):
+        mamba_mask = attention_mask
+
+        # No need for zeroing states when
+        # 1. Cached forward
+        # 2. Attending to all inputs
+        if cache_position[0] > 0 or (attention_mask is not None and torch.all(attention_mask == 1)):
+            mamba_mask = None
+
+        return mamba_mask
```

Comment:
Made this a separate function to make it more clean. Can be moved to the forward call directly too tho.

Reply:
sounds good, let's put the comments in the doc

Reply:
In which docs exactly? The arguments docs for the attention mask?

Reply:
I mean this:
```python
    def _update_mamba_mask(self, attention_mask, cache_position):
        """
                # No need for zeroing states when
                # 1. Cached forward
                # 2. Attending to all inputs
        """
        mamba_mask = attention_mask
        if cache_position[0] > 0 or (attention_mask is not None and torch.all(attention_mask == 1)):
            mamba_mask = None
        return mamba_mask
```